docs/modules/ROOT/pages/explanations/decisions/api-design.adoc

```diff
@@ -0,0 +1,250 @@
+For example all implementations for PostgreSQL have exactly the same API for the user.
+Due to the nature of the product, not all implementations provide the same functionality.
+The set of available features is most likely not the same between most providers.
+Also with Crossplane we're a bit limited on what's possible with translating the source API spec into the target API spec:
+
+* No conditionals in compositions
+* Only a limited set of transformations
+* Can't create random values
+
+[NOTE]
+This is not a complete list of limitations.
+
+== Requirements
+
+* APIs for a set of similar services feel similar
+* Only minimal configuration is needed for a default case
+* Unique features of a service provider can be leveraged
+
+== Proposals
+
+=== Proposal 1: Use same XRD with provider specific fields
+
+There's exactly one XRD for a set of services.
+This XRD contains a minimal set of common parameters, as well as a field for provider specific configurations.
+
+.Example Claim
+[source, yaml]
+----
+apiVersion: appcat.vshn.io/v1
+kind: PostgreSQL
+metadata:
+  name: my-instance
+spec:
+  parameters:
+    name: my-instance
+    majorVersion: '14'
+    pgSettings: {}
+    exoscale:
+      plan: hobbyist-2
+      type: pg
+      maintenance:
+        day: 2
+        hour: 22
+        minute: 0
+      backup:
+        hour: 20
+        minute: 0
+    vshn:
+      minorVersion: '4'
+----
+
+Advantages::
+
+* We can use the same XRD
+* Adding new providers to the list is easy
+
+Disadvantages::
+
+* We don't really have a "common look"
+* The customer needs to know what the underlying service provider is, even for simple instances
+
+=== Proposal 2: Separate XRDs
+
+This approach will have different XRDs for each provider.
+However, they will follow the same structure, as far as possible.
+All the toplevel fields will be the same, the differences will be apparent in the sub-fields.
+They will also have the same kind, but a different APIVersion.
+
+.Example Claim
+[source, yaml]
+----
+# Exoscale example
+apiVersion: exoscale.appcat.vshn.io/v1
+kind: PostgreSQL
+metadata:
+  name: my-instance
+spec:
+  parameters:
+    name: my-instance
+    service:
+      version: '14'
+      type: pg
+      settings: {}
+    size:
+      plan: hobbyist-2
+    maintenance:
+      day: 2
+      hour: 22
+      minute: 0
+    backup:
+      hour: 20
+      minute: 0
+
+# Converged example
+apiVersion: vshn.appcat.vshn.io/v1
+kind: PostgreSQL
+metadata:
+  name: my-instance
+spec:
+  parameters:
+    name: my-instance
+    service:
+      version: '14.4'
+      settings: {}
+      pgbouncer:
+        settings: {}
+    size:
+      cpu: 2
+      memory: 8Gi
+      disk: 200Gi
+      encrypted: true
+    maintenance:
+      day: 2
+      hour: 22
+      minute: 0
+    backup:
+      hour: 20
+      minute: 0
+----
+
+Advantages::
+
+* Same look between all providers, deploying PostgreSQL to any backend feels familiar
+* Specifics can be handled in a sensible way
```

Comment:
I wouldn't focus on "same look" too much. The same look can only apply if various providers share a common set of parameters. But then you'd have to agree on the same terminology. But depending on provider, some term might deviate from the DSL for the provider, maybe even mean the opposite.
As soon as one provider doesn't implement the "common" parameters then this "common look" is already outdated and sets expectations for the user only to be disappointed.

Reply:
See above comment. Maybe it's not quite clear what I mean with "common set of parameters". Think of them more of sets of more specific parameters.

For example most providers allow you to configure the service to some degree (version or settings). My proposal now states that all those params concerning the configuration of the service should go under `service`. Then, as a user, you know what these params are for. Same thing for the sizing or backups, etc. that might be common for a given service.

If a provider doesn't implement a given set of params at all, then we omit it.

Reply:
I have no strong feelings about grouping some provider-specific settings into top-level parameter groups. But I would avoid having those groups present in the API if they don't apply. I would also avoid renaming fields that are not matching the provider's naming and terminology.
If you think having some same-named group across different DB services suffices as having a "common look", then that's fine by me.